Semantics.md

```diff
@@ -678,6 +829,51 @@ outside the range which rounds to an integer in range) traps.
     This trap is intended to be impossible for user code to catch or handle, even in the future when it may be possible to
     handle some other kinds of traps or exceptions.
 
+## Thread operators
+
+  * `is_lock_free`: Given an operand `N`, if the atomic step of an atomic
+    primitive (see [Atomic Memory Accesses](#atomic-memory-accesses)) on a
+    datum of size `N` bytes will be performed without the agent acquiring a
+    lock outside the `N` bytes comprising the datum, then return `1`.
+    Otherwise, return `0`.
```

Comment:
Is `N` an immediate? It's not clear.

What type?

Are there invalid values for `N`?

Can we express the `constexpr`-ness of this? SAB had some wording somewhere about not changing between visible agents or something.

Are `wait` and `wake` atomic primitives? Because this wording would affect them and they definitely aren't lock-free. Currently the headings seem to say "no", it should remain that way.